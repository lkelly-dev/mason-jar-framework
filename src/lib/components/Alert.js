import React from 'react';
import { cx } from 'emotion';
import MODIFIERS from './Modifiers';

const Alert = props => {
  const { children, color, className, ...rest } = props;
  const alertColor = MODIFIERS.COLOR[color] ? `alert-${color}` : null;
  const alertClass = cx('alert', alertColor, className);
  return (
    <div role="alert" className={alertClass} {...rest}>
      {children}
    </div>
  );
};

const AlertLink = props => {
  const { children, className, ...rest } = props;
  const alertLinkClass = cx('alert-link', className);
  return (
    <a className={alertLinkClass} {...rest}>
      {children}
    </a>
  );
};

const AlertHeading = props => {
  const { children, className, ...restProps } = props;
  const alertHeadingClass = cx('alert-heading', className);
  return (
    <h4 className={alertHeadingClass} {...restProps}>
      {children}
    </h4>
  );
};

Alert.Link = AlertLink;
Alert.Heading = AlertHeading;

export { Alert, AlertLink, AlertHeading };
