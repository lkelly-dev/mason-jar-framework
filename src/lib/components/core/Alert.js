import React from 'react';
import { cx } from 'emotion';
import MODIFIERS from './Modifiers';

const Alert = props => {
  const { children, color, className } = props;
  const alertColor = MODIFIERS.COLOR[color] ? `alert-${color}` : null;
  const alertClass = cx('alert', alertColor, className);
  return (
    <div role="alert" className={alertClass}>
      {children}
    </div>
  );
};

const AlertLink = props => {
  const { children, className, ...restProps } = props;
  const alertLinkClass = cx('alert-link', className);
  return (
    <a className={alertLinkClass} {...restProps}>
      {children}
    </a>
  );
};

const AlertHeading = props => {
  const { children, className, ...restProps } = props;
  const alertHeadingClass = cx('alert-heading', className);
  return (
    <h4 className={alertHeadingClass} {...restProps}>
      {children}
    </h4>
  );
};

Alert.Link = AlertLink;
Alert.Heading = AlertHeading;

export { Alert, AlertLink, AlertHeading };
