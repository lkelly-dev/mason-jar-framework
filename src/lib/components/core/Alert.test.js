import React from 'react';
import { shallow } from 'enzyme';
import { Alert, AlertLink, AlertHeading } from "./Alert";

describe('Alert', () => {
  it('matches snapshot', () => {
    const component = shallow(<Alert />);
    expect(component).toMatchSnapshot();
  });

  it('contains children', () => {
    const component = shallow(<Alert><div>alert!</div></Alert>);
    expect(component.contains(<div>alert!</div>)).toEqual(true);
  });

  it('has className alert', () => {
    const component = shallow(<Alert />);
    expect(component.props().className).toBe("alert");
  });

  it('appends additional className', () => {
    const component = shallow(<Alert className="monkey" />);
    const classNameArray = component.prop('className').split(' ');
    expect(classNameArray).toContain('alert');
    expect(classNameArray).toContain('monkey');
  });
});

describe('Alert.Link', () => {
  it('matches snapshot', () => {
    const component = shallow(<AlertLink/>);
    expect(component).toMatchSnapshot();
  });

  it('contains children', () => {
    const component = shallow(<AlertLink><div>alert!</div></AlertLink>);
    expect(component.contains(<div>alert!</div>)).toEqual(true);
  });

  it('has className alert-link', () => {
    const component = shallow(<AlertLink />);
    expect(component.props().className).toBe("alert-link");
  });

  it('appends additional className', () => {
    const component = shallow(<AlertLink className="monkey" />);
    const classNameArray = component.prop('className').split(' ');
    expect(classNameArray).toContain('alert-link');
    expect(classNameArray).toContain('monkey');
    expect(classNameArray).toHaveLength(2);
  });
});

describe('Alert.Heading', () => {
  it('matches snapshot', () => {
    const component = shallow(<AlertHeading/>);
    expect(component).toMatchSnapshot();
  });

  it('contains children', () => {
    const component = shallow(<AlertHeading><div>headingtext</div></AlertHeading>);
    expect(component.contains(<div>headingtext</div>)).toEqual(true);
  });

  it('has className alert-link', () => {
    const component = shallow(<AlertHeading />);
    expect(component.props().className).toBe("alert-heading");
  });

  it('appends additional className', () => {
    const component = shallow(<AlertHeading className="monkey" />);
    const classNameArray = component.prop('className').split(' ');
    expect(classNameArray).toContain('alert-heading');
    expect(classNameArray).toContain('monkey');
    expect(classNameArray).toHaveLength(2);
  });
});
