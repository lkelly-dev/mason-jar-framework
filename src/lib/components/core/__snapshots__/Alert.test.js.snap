// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert matches snapshot 1`] = `
<div
  className="alert"
  role="alert"
/>
`;

exports[`Alert.Heading matches snapshot 1`] = `
<h4
  className="alert-heading"
/>
`;

exports[`Alert.Link matches snapshot 1`] = `
<a
  className="alert-link"
/>
`;
